x-pack/plugin/sql/qa/src/main/resources/nested.csv-spec

```diff
@@ -106,23 +106,23 @@ SELECT first_name f, last_name l, YEAR(dep.from_date) start FROM test_emp WHERE
 
 f:s                  | l:s           | start:i
 
-Sreekrishna          |Servieres      |1985           
-Zhongwei             |Rosen          |1986           
-Chirstian            |Koblick        |1986           
-null                 |Chappelet      |1988           
-Zvonko               |Nyanchama      |1989     
+Parto                |Bamford        |1995
+Chirstian            |Koblick        |1986
+Duangkaew            |Piveteau       |1996
+Kazuhide             |Peha           |1992
+null                 |Demeyer        |1994
 ;
 
 selectWithScalarOnNestedWithoutProjection
 SELECT first_name f, last_name l FROM test_emp WHERE dep.dep_name = 'Production' AND languages > 1 ORDER BY YEAR(dep.from_date) LIMIT 5;
 
 f:s                  | l:s
 
-Sreekrishna          |Servieres      
-Zhongwei             |Rosen          
-Chirstian            |Koblick        
-null                 |Chappelet      
-Zvonko               |Nyanchama  
+Parto                |Bamford
```

Comment:
This test is not relevant (anymore). Can you add `dep.from_date` to the projections?
Why did the order change since the PR should address ONLY the case of ordering by year and something else?

Reply:
The order can change because previously we would have a strict order on the whole `from_date` (including day and month) but now simply on the YEAR, so (apart from the YEAR) can be affected by the order of the doc entries when inserted to the index.

Reply:
Ok, so how does the query dsl look like now? Previously I think it contained a `"sort" : [{"from_date" : {"order" : "asc"}}]` section. We don't have tests that check the generated ES query with a sort by a date field?

Reply:
As discussed, since a scalar is used, it needs to be translated into a scripted sorting, which currently is not supported for nested fields.